### DISCUSSION TITLE
How to move pod to other available zone on GKE?

### DISCUSSION THREAD

**Discussion by clhlc** [strimzi/strimzi-kafka-operator#7648] [Q&A]
Hi, I have some question in My cluster.

I use Kafka (yaml) create kafka and zk cluster on gke cluster in (b,c,f)zone.
Now I create anther node pool in a-zone, So I want to transfer kafka and zk cluster to a-zone thouth Kafka(node affinity) https://strimzi.io/docs/operators/latest/configuring.html#affinity-str

Kafka data 2G, How can I to do? And how to transfer data to a-zone?

**Comment by clhlc** [comment]
I found sts updateStrategy is OnDelete, But I apply kafka.yaml, pod is auto update. I find k8s doc (ondelete) not auto delete nead manually delete pods

k8s docs:
https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#on-delete

<img width="288" alt="image" src="https://user-images.githubusercontent.com/10704095/202643497-4cfa8919-11a5-4b61-b853-974c6e59afe3.png">

**Comment by scholzj** [comment]
That depends on many aspects - how does your infrastructure look like, what storage you use and whether it can be used across zones etc., how exactly does your configuration look like etc. The affinity rules are only one of the ways how the scheduling is affected. It also for example follows the storage in case the PV cannot be moved. So depending on that, you might need to set up your new affinity rules and then delete the pod together with the PVC (either manually with `kubectl` or there is an annotation for it). Once they are deleted, it will be recreated with new PVC, bind to new volume and that might allow it to start in another zone. But as I said, it depends on many different aspects.
